include/onnxruntime/core/graph/graph.h

```diff
@@ -1106,6 +1106,14 @@ class Graph {
   // Graph value_info.
   std::vector<const NodeArg*> value_info_;
 
+  // Strings which have been used as node names.
+  // New node name should not conflict with this set.
+  std::unordered_set<std::string> generated_node_names_;
```

Comment:
>generated_node_names_ [](start = 34, length = 21)

some models seem to have no node names. Will that still be supported?

Reply:
If the function is called to generate names form them, yes. For example, it will not generate `""` if `""` is a node name in the model file.